tensorflow/python/keras/engine/training.py

```diff
@@ -944,10 +945,11 @@ def fit(self,
           epoch_logs.update(val_logs)
 
         callbacks.on_epoch_end(epoch, epoch_logs)
+        training_logs = epoch_logs
         if self.stop_training:
           break
 
-      callbacks.on_train_end()
+      callbacks.on_train_end(training_logs)
```

Comment:
```suggestion
      callbacks.on_train_end(logs=training_logs)
```

This way even reordering the arguments will keep the semantics